packages/web/lib/page.js

```diff
@@ -1,78 +1,78 @@
 "use strict";
 
-const {
-  resolveInternal,
-  resolveFn,
-  resolveFnScope,
-  resolveInternalScope,
-  validateScope,
-} = require("@browselang/core/lib/scope");
-
-const assertPageScope = (scope, message) => {
-  if (!validateScope((scope) => scope.internal.isPage, scope)) {
-    throw new Error(message);
-  }
-};
-
+const fs = require("fs-extra");
+const { resolveRule, resolveRuleScope } = require("@browselang/core/lib/scope");
 const { evalRule, evalRuleSet, getNewScope } = require("@browselang/core");
 const { help } = require("@browselang/core/lib/utils");
-const fs = require("fs-extra");
-const { keys } = require("./constants");
+const scope = require("@browselang/core/lib/scope");
 
-const dataStorageFn = (jsProcessing, type, optional = false) => (scope) => (
-  _opts
-) => async (key, selector) => {
-  assertPageScope(scope, `Cannot call @${type} outside of page context`);
-  const value = await resolveInternal("page", scope).$eval(
-    selector,
-    jsProcessing
-  );
-  if (value === null && !optional) {
-    throw new BrowseError({
-      message: `Element given to @${type} call had no content of the correct type. If this value is optional, try using @${type}?. This will set the value to null if a match isn't found`,
-      node: null,
-    });
-  }
-  const nearestPageScope = resolveInternalScope("page", scope);
-  nearestPageScope.internal.data[key] = value;
-  return value;
-};
-
-//Undefined is not understood by browse. Best practice is to always return null rather than undefined to make function conversion to browse easier later
+/**
+ * Data extraction functions
+ * Explicitly return `null` if no data is available
+ */
 const getString = (el) => el.textContent || el.innerText || null;
-
+const getStrings = (els) =>
+  els.map((el) => el.textContent || el.innerText || null).filter(Boolean);
 const getNumber = (el) => {
```

Comment:
Isn't part of this PR technically but this actually doesn't work if textContent is not a number but innerText is.